e2e/cypress/integration/channel/mark_as_unread_spec.js

```diff
@@ -0,0 +1,318 @@
+
+                // Another user creates posts in the channel since you can't mark your own posts unread currently
+                cy.postMessageAs({sender: sysadmin, message: 'post1', channelId: channelA.id}).then((p1) => {
+                    post1 = p1;
+
+                    cy.postMessageAs({sender: sysadmin, message: 'post2', channelId: channelA.id}).then((p2) => {
+                        post2 = p2;
+
+                        cy.postMessageAs({sender: sysadmin, message: 'post3', channelId: channelA.id, rootId: post1.id}).then((post) => {
+                            post3 = post;
+                        });
+                    });
+                });
+            });
+
+            cy.apiCreateChannel(teamId, 'channel-b', 'Channel B').then((resp) => {
+                channelB = resp.body;
+            });
+        });
+    });
+
+    afterEach(() => {
+        cy.apiDeleteChannel(channelA.id);
+        cy.apiDeleteChannel(channelB.id);
+    });
+
+    it('Channel should appear unread after switching away from channel and be read after switching back', () => {
+        // Starts unread
+        cy.get(`#sidebarItem_${channelA.name}`).should(beUnread);
+
+        switchToChannel(channelA);
+
+        // Then becomes read when you view the channel
+        cy.get(`#sidebarItem_${channelA.name}`).should(beRead);
+
+        markAsUnreadFromPost(post2);
+
+        // Then becomes unread when the channel is marked as unread
+        cy.get(`#sidebarItem_${channelA.name}`).should(beUnread);
+
+        switchToChannel(channelB);
+
+        // Then stays unread when switching away
+        cy.get(`#sidebarItem_${channelA.name}`).should(beUnread);
+
+        switchToChannel(channelA);
+
+        // And becomes read when switching back
+        cy.get(`#sidebarItem_${channelA.name}`).should(beRead);
+    });
+
+    it('New messages line should remain after switching back to channel', () => {
+        switchToChannel(channelA);
+
+        // Starts not visible
+        cy.get('.NotificationSeparator').should('not.exist');
+
+        markAsUnreadFromPost(post2);
+
+        // Then becomes visible
+        cy.get('.NotificationSeparator').should('exist');
+
+        switchToChannel(channelB);
+        switchToChannel(channelA);
+
+        // Then stays visible when switching back to channel
+        cy.get('.NotificationSeparator').should('exist');
+
+        switchToChannel(channelB);
+        switchToChannel(channelA);
+
+        // Then finally disappears when switching back a second time
+        cy.get('.NotificationSeparator').should('not.exist');
+    });
+
+    it('Should be able to mark channel as unread by alt-clicking', () => {
+        switchToChannel(channelA);
+
+        markAsUnreadFromPost(post2);
+
+        // The New Messages line should appear above the selected post
+        cy.get('.NotificationSeparator').should('exist');
+        cy.get('.NotificationSeparator').parent().parent().next().should('contain', 'post2');
```

Comment:
This can be chained so that getting `.NotificationSeparator` is done once only. You may also extract this into a function since it's used in several places.
```javascript
cy.get('.NotificationSeparator').
    should('exist').
    parent().parent().next().should('contain', post);
```